### DISCUSSION TITLE
RangeError: Maximum call stack size exceeded

### DISCUSSION THREAD

**Discussion by zrwsmd** [eclipse-langium/langium#1432] [Q&A] [answered]
when i define a langium file and execute npm run langium:generate,this error occured,I want to know what the reason is
![image](https://github.com/eclipse-langium/langium/assets/26995040/0a39845b-5057-48e5-9acd-69eb929a5970)

**Comment by msujew** [comment]
Can you share your grammar? The error you're seeing is likely related to this issue: https://github.com/eclipse-langium/langium/issues/1246.

**Reply by zrwsmd** [reply, marked as answer]
if i write like this 
entry st:
    'PROGRAM' name=Identifier
    (defenitions+=Definition)*
    (stStatements+=Statement_list)  
;
seems have this error 

but if i write like 
entry st:
    'PROGRAM' name=Identifier
    (stStatements+=Statement_list)
    (defenitions+=Definition)* 
;
it works well.
i guess it because * operator represent zero or many match,if it order first,can not decide it number,so report Maximum call stack size exceeded  error

**Reply by msujew** [reply]
Hm, that looks like a new issue. Note that I would recommend to do this instead:

```
(definitions+=Definition | statements+=Statement)*
```

Depending on your grammar of course. Limiting in which order rules can appear is often not a good idea.
